sdk/Trace/Sampler/TraceIdRatioBasedSampler.php

```diff
@@ -45,14 +48,21 @@ public function shouldSample(
         ?API\Attributes $attributes = null,
         ?API\Links $links = null
     ): SamplingResult {
+        // TODO: Add config to adjust which spans get sampled (only default from specification is implemented)
         if (null !== $parentContext && ($parentContext->getTraceFlags() & API\SpanContext::TRACE_FLAG_SAMPLED)) {
             return new SamplingResult(SamplingResult::RECORD_AND_SAMPLED, $attributes, $links);
         }
-
-        // TODO: implement as a function of TraceID when specification is ready
-        $decision = (lcg_value() < $this->probability)
-            ? SamplingResult::RECORD_AND_SAMPLED
-            : SamplingResult::NOT_RECORD;
+        /**
+         * Since php can only store up to 63 bit positive integers
+         */
+        $traceIdLimit = (1 << 60) - 1;
+        $lowerOrderBytes = hexdec(substr($traceId, strlen($traceId) - 15, 15));
+        $traceIdCondition = $lowerOrderBytes < round($this->probability * $traceIdLimit);
+        $decision = SamplingResult::NOT_RECORD;
+        // TODO: Also sample Spans with remote parent
```

Comment:
I am not sure how to find out if the current Span has a remote parent without its Span Context, might be related to the other issue requiring implementation of the IsRemote attribute.

Reply:
Yes - I believe we may have to implement the SpanContext `isRemote` attribute for this.  I think a TODO is sufficient here.  Bonus points for creating an issue for this once this PR is merged.

Reply:
Will do.